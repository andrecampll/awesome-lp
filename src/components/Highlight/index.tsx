import { Wrapper, Title, Subtitle, Content } from './styles';

import Button from '../Button';

export type HighlightProps = {
  title: string;
  subtitle: string;
  backgroundImage: string;
  buttonLabel: string;
  buttonLink: string;
};

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  backgroundImage,
}: HighlightProps) => (
  <Wrapper backgroundImage={backgroundImage}>
    <Content>
      <Title>{title}</Title>
      <Subtitle>{subtitle}</Subtitle>

      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </Content>
  </Wrapper>
);

export default Highlight;
