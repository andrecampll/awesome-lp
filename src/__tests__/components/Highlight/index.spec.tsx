import { screen } from '@testing-library/react';
import Highlight from '../../../components/Highlight';

import { renderWithTheme } from '../../../utils/tests/helpers';

const props = {
  title: 'heading 1',
  subtitle: 'heading 2',
  floatImage: '/float-image.png',
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2',
};

describe('Highlight', () => {
  it('should be able to render headings and button', () => {
    const { container } = renderWithTheme(<Highlight {...props} />);

    expect(
      screen.getByRole('heading', {
        name: /heading 1/i,
      }),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('heading', {
        name: /heading 2/i,
      }),
    ).toBeInTheDocument();

    expect(
      screen.getByRole('link', {
        name: /buy now/i,
      }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able to render background image', () => {
    const { container } = renderWithTheme(<Highlight {...props} />);

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`,
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be able to render background image', () => {
    renderWithTheme(<Highlight {...props} floatImage="/float-image.png" />);

    expect(
      screen.getByRole('img', {
        name: props.title,
      }),
    ).toHaveAttribute('src', '/float-image.png');
  });
});
