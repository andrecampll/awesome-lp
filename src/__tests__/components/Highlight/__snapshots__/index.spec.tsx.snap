// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight should be able to render background image 1`] = `
<section
  class="sc-bdfBwQ cGYIoj"
>
  <img
    alt="heading 1"
    class="sc-gsTCUz egLOHZ"
    src="/float-image.png"
  />
  <div
    class="sc-dlfnbm kXoxNa"
  >
    <h2
      class="sc-hKgILt llWGmG"
    >
      heading 1
    </h2>
    <h3
      class="sc-eCssSg hgfixj"
    >
      heading 2
    </h3>
    <a
      class="sc-jSgupP fQOmRy"
      href="/rdr2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`;

exports[`Highlight should be able to render headings and button 1`] = `
<section
  class="sc-bdfBwQ cGYIoj"
>
  <img
    alt="heading 1"
    class="sc-gsTCUz egLOHZ"
    src="/float-image.png"
  />
  <div
    class="sc-dlfnbm kXoxNa"
  >
    <h2
      class="sc-hKgILt llWGmG"
    >
      heading 1
    </h2>
    <h3
      class="sc-eCssSg hgfixj"
    >
      heading 2
    </h3>
    <a
      class="sc-jSgupP fQOmRy"
      href="/rdr2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`;
